src/types/instantsearch.ts

```diff
@@ -116,6 +116,11 @@ export interface Router<TRouteState = UiState> extends Widget {
    * return a string. It may return an empty string.
    */
   createURL(state: TRouteState): string;
+
+  /**
+   * Called when InstantSearch is disposed. Used to remove subscriptions.
+   */
+  dispose?(): void;
```

Comment:
We could make it required though since all the other methods are required.

Reply:
agreed, seems like a good method to make required

Reply:
https://github.com/algolia/instantsearch.js/pull/4120/commits/fe3e21315dea1f00bcc12b18158c8f8721553c11